packages/deployer/internal/initializable-ast-validator.js

```diff
@@ -0,0 +1,69 @@
+const {
+  findContractDefinitions,
+  findContractDependencies,
+  findFunctions,
+} = require('@synthetixio/core-js/utils/ast/finders');
+
+class ModuleInitializableASTValidator {
+  constructor(asts) {
+    this.contractNodes = Object.values(asts).map(findContractDefinitions).flat();
+  }
+
+  findMissingIsInitialized() {
+    const errors = [];
+
+    for (const contractName of this.findInitializableContractNames()) {
+      const functionName = `is${_capitalizeContractName(contractName)}Initialized`;
+      if (!findFunctions(contractName, this.contractNodes).some((v) => v.name === functionName)) {
+        errors.push({
+          msg: `Initializable contract ${contractName} missing ${functionName} function!}`,
+        });
+      }
+    }
+
+    return errors;
+  }
+
+  findMissingInitializer() {
+    const errors = [];
+
+    for (const contractName of this.findInitializableContractNames()) {
+      const functionName = `initialize${_capitalizeContractName(contractName)}`;
+      if (!findFunctions(contractName, this.contractNodes).some((v) => v.name === functionName)) {
+        errors.push({
+          msg: `Initializable contract ${contractName} missing ${functionName} function!}`,
+        });
+      }
+    }
+
+    return errors;
+  }
+
+  findInitializableContractNames() {
+    const initializableContractNames = [];
+
+    const contractNames = this.contractNodes.map((v) => v.name);
+
+    for (const contractName of contractNames) {
+      if (contractName === 'InitializableMixin') {
+        continue;
+      }
```

Comment:
Just realized, this is not compatbile when importing using `as`:
```solidity
import {InitializableMixin as Sarasa} from  ...
```

Reply:
You are right! I thought that we were just getting the inherited names, not finding the contract nodes by id: https://github.com/Synthetixio/synthetix-v3/blob/main/packages/core-js/utils/ast/finders.js#L143:L150

I'm closing the issue :)